### PR TITLE
Feat/contact change

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

 - Refactor contacts servcies section
- Do not allow contact billing change for services with a debt 

MBE-269

## Related 
- [x] https://github.com/ovh-ux/ovh-api-services/pull/220

⚠️ Texts are being reviewed by content team

### CODE REVIEW

**Comment by marie-j** [comment]
Edit: Content has been reviewed

**Comment by ovh-cds** [comment]
CDS Report build-manager#2467.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✔ 
  * Build manager US ✔ 
  * Tests ✘

**Comment by antleblanc** [comment]
@marie-j :tada: `ovh-api-services@9.4.0` has been released (https://github.com/ovh-ux/ovh-api-services/blob/master/CHANGELOG.md#940-2019-08-23)

File: client/app/account/contacts/service/contacts-service.service.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { AVAILABLE_SERVICES } from './user-contacts.constants';
+
+export default class {
+  /* @ngInject */
+  constructor(OvhApiOvhProduct, OvhHttp) {
+    this.OvhApiOvhProduct = OvhApiOvhProduct;
+    this.OvhHttp = OvhHttp;
+  }
+
+  // TODO: Find a way to inject ovh-api-services depending on the service category
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/contacts-service.service.js]: Looks like a good requirement for the successor of `ovh-api-services`, maybe you could add it to the document of requirements (ask JC for it as I don't know where it is :D)

File: client/app/account/contacts/service/contacts-service.service.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        contactAdmin: service.contactAdmin,
+        contactBilling: service.contactBilling,
+        contactTech: service.contactTech,
+      },
+    });
+  }
+
+  getServices() {
+    return this.OvhApiOvhProduct.Aapi().query().$promise
+      .then((services) => {
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/contacts-service.service.js]:
```suggestion
      .then((services) => _.map(
        _.filter(
          services,
          service => AVAILABLE_SERVICES.includes(service.category),
        ),
        service => new BillingService(service),
      )
```

File: client/app/account/contacts/service/edit/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "account_contacts_service_edit": "Modifier les contacts",
+  "account_contacts_service_edit_description": "Modifier les contacts du service {{ serviceName }}",
+  "account_contacts_service_edit_contact_admin": "Contact administrateur",
+  "account_contacts_service_edit_contact_tech": "Contact technique",
+  "account_contacts_service_edit_contact_billing": "Contact facturation",
+  "account_contacts_service_edit_cancel": "Annuler",
+  "account_contacts_service_edit_confirm": "Valider",
+  "account_contacts_service_edit_error": "Une erreur est survenue lors de la demande de changement de contact : {{ message }}",
+  "account_contacts_service_edit_success": "La demande de changement de contact a été effectuée avec succès. Les contacts concernés par la demande recevront un email contenant la procédure de validation.",
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/edit/translations/Messages_fr_FR.json]:
```suggestion
  "account_contacts_service_edit_success": "La demande de changement de contact a été effectuée avec succès. Les contacts concernés par la demande recevront un e-mail contenant la procédure de validation.",
```

File: client/app/account/contacts/service/edit/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  "account_contacts_service_edit_warn_debt": "Vous ne pouvez pas modifier le contact de facturation, car une facture est en attente de règlement pour ce service. Nous vous invitons à l’acquitter afin d’effectuer ce changement.",
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/edit/translations/Messages_fr_FR.json]:
```suggestion
  "account_contacts_service_edit_warn_debt": "Vous ne pouvez pas modifier le contact de facturation, car une facture est en attente de règlement pour ce service. Nous vous invitons à l’acquitter avant de pouvoir effectuer ce changement.",
```
?

File: client/app/account/contacts/service/edit/translations/Messages_fr_FR.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  "account_contacts_service_edit_warn_contact_billing": "Vous ne pouvez pas modifier le contact de facturation, car une facture est en attente de règlement pour ce service. Nous vous invitons à joindre le contact de facturation actuel afin qu’il effectue ce changement."
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/edit/translations/Messages_fr_FR.json]: Could we add the nic or a link to the contact page so they know who the billing contact is ?

File: client/app/account/contacts/service/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  "account_contacts_service_category_NASHA": "NAS-HA",
+  "account_contacts_service_category_VRACK": "vRACK",
+  "account_contacts_service_category_DOMAIN": "Domaine",
+  "account_contacts_service_category_EMAIL_DOMAIN": "Emails",
+  "account_contacts_service_category_CDN_WEBSTORAGE": "Cdn WebStorage",
+  "account_contacts_service_category_CDN_WEBSITE": "Cdn WebSite",
+  "account_contacts_service_category_CDN_DEDICATED": "Cdn dédié",
+  "account_contacts_service_category_PRIVATE_DATABASE": "Base de données privée",
+  "account_contacts_service_category_ZONE": "Zone DNS",
+  "account_contacts_service_category_LOAD_BALANCING": "IP load balancing",
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/translations/Messages_fr_FR.json]:
```suggestion
  "account_contacts_service_category_LOAD_BALANCING": "IP Load Balancing",
```
?

File: client/app/account/contacts/service/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  "account_contacts_service_category_SITE_BUILDER": "Site Builder",
+  "account_contacts_service_category_TELEPHONY": "Téléphonie",
+  "account_contacts_service_category_SMS": "SMS",
+  "account_contacts_service_category_VPS": "VPS",
+  "account_contacts_service_category_XDSL": "xDSL",
+  "account_contacts_service_category_NAS": "NAS",
+  "account_contacts_service_category_NASHA": "NAS-HA",
+  "account_contacts_service_category_VRACK": "vRACK",
+  "account_contacts_service_category_DOMAIN": "Domaine",
+  "account_contacts_service_category_EMAIL_DOMAIN": "Emails",
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/translations/Messages_fr_FR.json]:
```suggestion
  "account_contacts_service_category_EMAIL_DOMAIN": "E-mails",
```

File: client/app/account/contacts/service/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  "account_contacts_service_category_CDN_WEBSTORAGE": "Cdn WebStorage",
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/translations/Messages_fr_FR.json]:
```suggestion
  "account_contacts_service_category_CDN_WEBSTORAGE": "CDN WebStorage",
```
?

File: client/app/account/contacts/service/edit/translations/Messages_fr_FR.json (hunk 4 above)
Comment (marie-j) [reply client/app/account/contacts/service/edit/translations/Messages_fr_FR.json]: This is content text

**Comment by ovh-cds** [comment]
CDS Report build-manager#2502.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✘

**Comment by ovh-cds** [comment]
CDS Report build-manager#2502.2 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✘  
  * Build manager EU ✘  
  * Build manager US ✘  
  * Tests ✔

**Comment by ovh-cds** [comment]
CDS Report build-manager#2502.3 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✘

**Comment by jleveugle** [comment]
Hello @marie-j,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

File: client/app/models/BillingService.class.js [hunk 9]
```diff
@@ -65,7 +65,12 @@ export default class BillingService {
   }
 
   hasDebt() {
-    return _.includes(['PENDING_DEBT', 'UN_PAID'], this.status);
+    return _.includes([
+      'PENDING_DEBT',
```
Comment (FredericEspiau) [inline client/app/models/BillingService.class.js]:
````js
_.includes([
  'PENDING_DEBT',
  'UN_PAID',
],
upperCase(
  snakeCase(this.status)
))
````

File: client/app/account/contacts/service/translations/Messages_fr_FR.json [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  "account_contacts_service_category_VRACK": "vRACK",
+  "account_contacts_service_category_DOMAIN": "Domaine",
+  "account_contacts_service_category_EMAIL_DOMAIN": "E-mails",
+  "account_contacts_service_category_CDN_WEBSTORAGE": "CDN WebStorage",
+  "account_contacts_service_category_CDN_WEBSITE": "CDN WebSite",
+  "account_contacts_service_category_CDN_DEDICATED": "CDN dédié",
+  "account_contacts_service_category_PRIVATE_DATABASE": "Base de données privée",
+  "account_contacts_service_category_ZONE": "Zone DNS",
+  "account_contacts_service_category_LOAD_BALANCING": "IP Load Balancing",
+  "account_contacts_service_category_LOAD_BALANCER": "Ip Load Balancer",
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/translations/Messages_fr_FR.json]:
```suggestion
  "account_contacts_service_category_LOAD_BALANCER": "IP Load Balancer",
```

File: client/app/account/contacts/service/translations/Messages_fr_FR.json [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  "account_contacts_service_category_OVER_THE_BOX": "overTheBox",
```
Comment (FredericEspiau) [inline client/app/account/contacts/service/translations/Messages_fr_FR.json]:
```suggestion
  "account_contacts_service_category_OVER_THE_BOX": "OverTheBox",
```

?

File: client/app/models/BillingService.class.js (hunk 9 above)
Comment (cbourgois) [reply client/app/models/BillingService.class.js]: constants ?

File: client/app/account/contacts/service/edit/edit.module.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import component from './edit.component';
+import routing from './edit.routing';
+
+const moduleName = 'ovhManagerDedicatedAccountContactsServiceEdit';
+
+angular.module(moduleName, [
```
Comment (cbourgois) [inline client/app/account/contacts/service/edit/edit.module.js]:
```suggestion
angular.module(moduleName, [])
```

File: client/app/account/contacts/service/contacts-service.service.js (hunk 2 above)
Comment (antleblanc) [reply client/app/account/contacts/service/contacts-service.service.js]: Both solutions are correct. We can keep the default one.

**Pull request merged**